### PR TITLE
define registered domain and registrable domain; fixes #12

### LINKED ISSUE #12: define "registrable domain" "registered domain" on publicsuffix.org/list
the Algorithm section of https://publicsuffix.org/list/ contains this step:
`7.  The registered or registrable domain is the public suffix plus one additional label.`
..but "registered domain" and "registrable domain" are not defined. 

Discussion with `<team@publicsuffix.org>` yielded these definitions:
> 7. The registered (or registrable) domain is the public suffix plus one additional label, where:
> * A registered domain is a domain that is registered, which is a process defined within the domain name system related to IANA functions, ICANN, and domain name registries.
> * A registrable domain is a formally valid domain name that can be registered at a registry, i.e., it is not presently registered. 

Perhaps the above definitions (or refinements thereof) could be added to https://publicsuffix.org/list/ ?   various folks are using the term "registrable domain" in various specs where they likely mean "registered domain" (but the original text of step 7 made it easier to latch onto the term "registrable domain" by default). 

Please see also PR #2.

[this issue supersedes publicsuffix/list#236]

### PR DESCRIPTION
**Pull request by equalsJeffH**
patch resolving #12 (see also: publicsuffix/list#236) submitted per suggestion.

### CODE REVIEW

File: list/index.html [hunk 1]
```diff
@@ -185,7 +185,17 @@ <h2 id="list-format">
                 The public suffix is the set of labels from the domain which match the labels of the prevailing rule, using the matching algorithm above.
             </li>
             <li>
-                The registered or registrable domain is the public suffix plus one additional label.
+                The registered or registrable domain is the public suffix plus one additional label, where..
```
Comment (gerv) [inline list/index.html]: ".." should be ":"?

Comment (equalsJeffH) [inline list/index.html]: sure, using a colon there would be fine. thanks.

**Comment by equalsJeffH** [comment]
hm. actually, in thinking about this more -- the impetus being https://github.com/whatwg/html/pull/2365 -- I'm inclined to modify these proposed changes to be..
> 7. The registered (or registrable) domain is the public suffix plus one additional label, where:
> * A registered domain is a domain that is registered, which is a process defined within the domain name system related to IANA functions, ICANN, and domain name registries.
> * A registrable domain is a formally valid domain name that could be registered at a registry given the state of the public suffix list at the time this algorithm was run, although it may or may not be presently registered.

**Comment by gerv** [comment]
> Another term occasionally used for a registered/registrable domain is "effective Top Level Domain plus one (eTLD+1)"

No; we are trying very hard not to use that terminology, and renamed the PSL and its file specifically to get away from it. Various people in the domain world think the idea of an "effective TLD" is unhelpful and confusing, and they are right.

**Comment by gerv** [comment]
The rest seems OK.

**Comment by equalsJeffH** [comment]
wrt @gerv's pushback on this particular suggested (last) bullet item in https://github.com/publicsuffix/publicsuffix.org/pull/2#issuecomment-281520351:
> Another term occasionally used for a registered/registrable domain is "effective Top Level Domain plus one (eTLD+1)"

wherein @gerv said:
> No; we are trying very hard not to use that terminology, and renamed the PSL and its file specifically to get away from it. Various people in the domain world think the idea of an "effective TLD" is unhelpful and confusing, and they are right.

I (and some other folks I've asked about this) respectfully disagree with that position. The "eTLD" and "eTLD+1" terms have acknowledged issues but they have been, and continue to be, used in a number of places for various reasons and regardless of the PSL's attempt to "disappear" the terms.  We feel that it is more helpful for the community to acknowledge the use of those terms in the context of defining new ones. Thus folks searching for the "old" terms will/may come across this page <https://publicsuffix.org/list/> and see them being deprecated "in context".  Not mentioning the old terms here just means that folks searching for them or their definitions more likely will not come across <https://publicsuffix.org/list/> and thus be guided towards using the new terms. 

With that in mind, here's a suggested revision to that last bullet item:
* Another term occasionally used for a registered/registrable domain is "effective Top Level Domain plus one (eTLD+1)". However, this latter term is deprecated and we encourage you to instead use "registrable domain" instead.

**Comment by gerv** [comment]
What are the "various reasons"?

"Effective Top Level Domain" is misleading, because a public suffix doesn't behave like a TLD in some circumstances, and doesn't even necessarily have open registration (which is what prompted the eTLD name in the first place).

**Comment by equalsJeffH** [comment]
> What are the "various reasons"?

Because the term (eTLD) is already in wide use (for better or worse) and folks use what they are familiar with, because there's not a succinct well-known acronym for e.g. "public suffix + 1" (as opposed to "eTLD+1"), etc.

I am not advocating that folks continue to use "Effective Top Level Domain" (eTLD+1). Rather, I am trying to suggest that having search results for either of the foregoing include publicsuffix.org/list/ wherein the terms are defined and eTLD is deprecated is preferrable to trying to simply "disappear" the latter term by not mentioning it (ain't gonna work).

**Comment by sleevi** [comment]
To add more fuel to the fire of debate:
https://webkit.org/blog/7675/intelligent-tracking-prevention/ calls them "privately-controlled domain or TLD+1"

**Comment by weppos** [comment]
I like the current content of the patch, as in the context of this project uses the "public suffix" work which has a specific meaning.

However, as a matter of fact, I've personally been using the term eTLD quite frequently in most of the projects I've been working on. As @equalsJeffH it's a short, concise term that applies quite well to code and documentation, especially when you have to distinguish between TDL vs eTLD.

For example, as a registrar, we have to store both information and use them both depending on who we interact with, and what is the purpose of the validation.

In the past I used suffix, publicsuffix, and various other terms, but each time the terms were too confusing, ambiguous (especially in technical contexts where `registerable domain` may have dozens of different meanings). eTLD is "forged name", and the only people I've seen confused are the ones that ignore the existence of the PSL itself (that would be confused otherwise).

If we want to analyze the semantic of the term, I agree that saying that `github.io` is an _effective top level domain_ may be misleading because it's not open for registration as opposite to e.g `co.uk`, but it's also true that if we consider eTLD as an invented word (pretty much like it would be `fafa`) then as long as you are taught that `fafa` == `a public suffix`, then it's pretty easy to digest.

I trust your decision @gerv and I won't endorse the use of eTLD here, I just wanted to provide a personal feedback as one that has been using this term (and reading this term) quite frequently and sucessfully.

**Comment by equalsJeffH** [comment]
The terms "effective top-level domain" (eTLD) and "effective top-level domain+1"  (eTLD+1)  [continue to be used fairly widely](https://tools.ietf.org/html/draft-ietf-tokbind-https-15), hence [as noted above](https://github.com/publicsuffix/publicsuffix.org/pull/2#issuecomment-308255520), efforts to simply "disappear" them would seem to be a disservice to "the community". 

I continue to suggest that the term's existence and definition ought to be acknowledged somewhere "official", along with other widely-used synonymous terms and variations such that there exists a "rosetta stone" of sorts.  

This relatively simple PR is a modest step in that direction.

**Comment by weppos** [comment]
Thanks for the feedback @equalsJeffH. I'd actually like to split the discussion about eTLD and temporary resurrect this PR for the following reason:

I was investigating the changes in this PR, and I see that nowhere there is a mention to eTLD. In fact, the only attempt of this PR is to clarify the meaning of "registered domain" and "registerable domain".

@sleevi @dnsguru do you have any specific objection regarding the definition provided by @equalsJeffH in this PR about _registered domain_ and _registerable domain_?

If not, I propose to:

1. Merge the PR
2. Move the discussion about eTLD in the mailing list

**Comment by dnsguru** [comment]
@weppos, @sleevi  I concur on the proposed outcomes 1 & 2

Comment: @equalsJeffH makes a compelling argument for the inclusion of the eTLD and eTLD +1 being at least defined so that someone searching can learn more and put those acronyms or their full wording in context of how we are defining them, it improves clarity and reduces confusion.

One of the largest challenges I face when sitting in meetings with ccTLD or gTLD registries or some of the developer entities that have private section entries is that people's views are quite lenticular and based upon their subjective/perspective needs.   

Stated more plainly, if you asked ten people what an 'effective top-level domain' was, you might get ten answers with variation significant enough to cause revisions to submissions vs correct initial submissions.  In purely selfish terms as a volunteer, it takes time to clarify if we're not clear in writing.  Ain't nobody got time for that.  We should be helping it be more clear.

**Comment by weppos** [comment]
@dnsguru let's put aside the eTLD definition for now. Would you and @sleevi be satisfied with the current PR changes as a definition of registerable domain?

**Comment by sleevi** [comment]
@weppos As we were recently discussing elsewhere, I don't think this definition fully captures what the PSL is/does. For example, I don't think `foo.appspot.com` meets either of these proposed definitions - do you?

**Comment by weppos** [comment]
@sleevi I concur it doesn't cover the cases in the PRIVATE section. The question is whether we want to:

1. Try to find a single definition for both
2. Split the definition and then keep registered/registerable domains, then integrate with a new concept that covers the appspot.com

PS. I'm sorry if I rely on both of you, but not being an English speaker I'm afraid I may miss the necessary language mastery to suggest a valid alternative.

**Comment by sleevi** [comment]
I'm not sure the suggested alternative, but I realize the remark I just added is effectively @gerv's argument in https://github.com/publicsuffix/publicsuffix.org/pull/2#issuecomment-305158841 about why eTLD/eTLD+1 is a bad term, and I agree with that.

So I don't think we should take this PR as is, because it lacks that nuance, and would only further encourage the eTLD mis-moniker.

**Comment by weppos** [comment]
If we don't clarify what a registered/registerable domain is, then we probably need to remove these terms as part of the section and find a better way to describe that concept.

Using those words and pretend that any reader has the same interpretation we have it's confusing.

**Pull request closed without merging**